src/status_im/ui/screens/wallet/transactions/events.cljs

```diff
@@ -21,39 +21,35 @@
-      (let [{:keys [from to value data gas gasPrice]} args]
-        (let [;;TODO (andrey) revisit this map later (this map from old transactions, idk if we need all these fields)
-              transaction {:id         id
-                           :from       from
-                           :to         to
-                           :value      (money/to-decimal value)
-                           :data       data
-                           :gas        (money/to-decimal gas)
-                           :gas-price  (money/to-decimal gasPrice)
-                           :timestamp  now
-                           :message-id message_id}]
-          (merge
-            {:db (-> db
-                     (assoc-in [:wallet :transactions-unsigned id] transaction)
-                     (assoc-in [:wallet/send-transaction :transaction-id] id))}
-            (if (:waiting-signal? send-transaction)
-              ;;sending from wallet
-              {:dispatch [:wallet.send-transaction/transaction-queued id]}
-              ;;sending from chat
-              {:dispatch [:navigate-to-modal :wallet-send-transaction-modal {:amount         (str (money/wei->ether value))
-                                                                             :transaction-id id
-                                                                             :to-address     to
-                                                                             :to-name        to
-                                                                             :from-chat?     true}]}))))
+      (let [{:keys [from to value data gas gasPrice]} args
+            ;;TODO (andrey) revisit this map later (this map from old transactions, idk if we need all these fields)
+            transaction                               {:id         id
+                                                       :from       from
+                                                       :to         to
+                                                       :value      (money/to-decimal value)
+                                                       :data       data
+                                                       :gas        (money/to-decimal gas)
+                                                       :gas-price  (money/to-decimal gasPrice)
+                                                       :timestamp  now
+                                                       :message-id message_id}]
+        (merge
+         {:db (-> db
+                  (assoc-in [:wallet :transactions-unsigned id] transaction)
+                  (assoc-in [:wallet :send-transaction :id] id))}
+         (if (get-in db [:wallet :send-transaction :waiting-signal?])
+           ;;sending from wallet
+           {:dispatch [:wallet.send-transaction/transaction-queued id]}
+           ;;sending from chat
+           {:dispatch [:navigate-to-modal :wallet-send-transaction-modal id :from-chat]})))
       {:discard-transaction id})))
```

Comment:
Why is `:from-chat?` not set anymore when signal from go about queued transaction arrives ? It's necessary so closing modal correctly  removes transaction from unsigned-queue (and discards it).

Reply:
thanks I was going to investigate why it doesn't close anymore as you can see in my comment above